### PR TITLE
docs: add architecture and design document

### PR DESCRIPTION
**Pull request by rjayasinghe**
## Summary

- Adds `docs/architecture.md` documenting the plugin's overall design
- Covers the CAP Java messaging core abstractions the plugin builds on (`AbstractMessagingService`, `BrokerConnectionProvider`, `BrokerConnection`)
- Describes the boot flow, two-binding model (AEM broker + validation service), AMQP/OAuth2 connection mechanics, SEMP v2 management plane, validation handshake, and configuration surface
- Includes a comparison table against `cds-feature-enterprise-messaging`
- Lists caveats and edge cases identified during analysis

## Test plan

- [ ] Review document for accuracy against current source
- [ ] Verify all file path and line number references are still valid

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+---
+
+## 4. Two binding types, two roles
+
+The plugin reads **two** distinct service bindings:
+
+| Binding        | Tag/Name                  | Purpose                                                                  | Auth                                                            |
+|----------------|---------------------------|--------------------------------------------------------------------------|-----------------------------------------------------------------|
+| AEM broker     | `advanced-event-mesh`     | AMQP+JMS data plane *and* SEMP management plane                          | OAuth2 client_credentials against IAS (`authentication-service`) |
+| Validation     | `aem-validation-service`  | One-time handshake confirming the VMR is provisioned for this subaccount | OAuth2 against XSUAA (`handshake.oa2`)                          |
```
Comment (t-bonk) [inline docs/architecture.md]: The goal of this validation is to check whether the AEM service is resold by SAP (reply with HTTP 20x) or the customer sourced it directly from Solace (HTTP > 399).

File: docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+Notably, AEM does **not** override `toFullyQualifiedTopicName` or `toFullyQualifiedQueueName` like EM does. EM has a non-trivial namespace story (`$namespace` placeholders, `+/+/+/` wildcards for cross-tenant subscriptions, `.`→`/` translation for CloudEvents). AEM treats topics as opaque strings; multi-tenant fan-out and namespace conventions are the user's problem.
+
+---
+
+## 8. Validation handshake
+
+`AemValidationClient.validate(managementUri, subaccountId)` POSTs `{"hostName": <host extracted from managementUri>, "subaccountId"?: ...}` to the validation service's handshake endpoint (the destination base URI from the validation binding). Only the *host* component of `managementUri` is sent, not the full URI. It runs at most once per service instance — guarded by the `aemBrokerValidated` volatile `Boolean` flag in `AemMessagingService.validate()` (line 165), and only on the first publish (lazily via `emitTopicMessage`). A 200 confirms this BTP subaccount is entitled to this VMR. Anything else → `ServiceException`, no message is sent.
+
+This is a compliance / anti-misconfiguration check: it prevents an app from being mistakenly bound to a broker in the wrong subaccount.
```
Comment (t-bonk) [inline docs/architecture.md]: Wrong! See above

**Review by t-bonk — CHANGES_REQUESTED** [review]
All in all it LGTM. There is only one change that crucial. Also please remove all line references as the line numbers might change when fixing issue or adding features.

**Comment by rjayasinghe** [comment]
> All in all it LGTM. There is only one change that crucial. Also please remove all line references as the line numbers might change when fixing issue or adding features.

@t-bonk Done the requested changes. Please check again :)
